### PR TITLE
Issue 116: Add function to back calculate Rt

### PR DESCRIPTION
**Pull request by seabbs**
This started as a PR for #165 (hence the branch name) but is just addressing (and closing #116) as #165 is turning out to be a little more complex than expected. 

Here we add two new functions

- `expected_Rt` Dispatching on a `EpiData` object or taking arguments directly.
- `expected_rt` dispatching on a vector and taking the log differences.

### CODE REVIEW

**Comment by seabbs** [comment]
Closing as we. are now planing on doing this within the inf submodel

**Comment by seabbs** [comment]
Okay looking at this again I vote to review and merge as a partial PR towards #116 leaving the expected growth rate to work out

File: EpiAware/src/EpiInfModels/EpiData.jl [hunk 1]
```diff
@@ -61,3 +61,36 @@ struct EpiData{T <: Real, F <: Function}
...
+- `exp_Rt::Vector{Float64}`: A vector of expected Rt values.
+
+## Examples
+
+```julia
+using EpiAware
+
+data = EpiData([0.2, 0.3, 0.5], exp)
+infections = [100, 200, 300, 400, 500]
+expected_Rt(data, infections)
```
Comment (SamuelBrand1) [inline EpiAware/src/EpiInfModels/EpiData.jl:86]: close the example block

File: EpiAware/src/EpiInfModels/EpiData.jl [hunk 2]
```diff
@@ -61,3 +61,36 @@ struct EpiData{T <: Real, F <: Function}
...
+# Arguments
+- `data::EpiData`: An instance of the EpiData type containing generation interval data.
+- `infections::Vector{<:Real}`: A vector of infection data.
+
+# Returns
+- `exp_Rt::Vector{Float64}`: A vector of expected Rt values.
+
+## Examples
+
+```julia
```
Comment (SamuelBrand1) [inline EpiAware/src/EpiInfModels/EpiData.jl:81]:
```suggestion
```julia
using EpiAware
data = EpiData([0.2, 0.3, 0.5], exp)
infections = [100, 200, 300, 400, 500]
expected_Rt(data, infections)
```
```

File: EpiAware/src/EpiInfModels/EpiData.jl [hunk 3]
```diff
@@ -61,3 +61,36 @@ struct EpiData{T <: Real, F <: Function}
...
+using EpiAware
+
+data = EpiData([0.2, 0.3, 0.5], exp)
+infections = [100, 200, 300, 400, 500]
+expected_Rt(data, infections)
+"
+function expected_Rt(data::EpiData, infections::Vector{<:Real})
+    n = data.len_gen_int
+
+    denom_Rt = [dot(reverse(data.gen_int),
```
Comment (SamuelBrand1) [inline EpiAware/src/EpiInfModels/EpiData.jl:91]: Might be worth having an `@assert` on length of infections vs `n`

**Review by SamuelBrand1 — CHANGES_REQUESTED** [review]
LGTM. One docstring small fix.

Worth noting that this will give the _effective_ $R_t$ as opposed to unadjusted $R_t$; this will diverge as we add population effects e.g. #245 . This is a good thing but maybe also mention in docstring.

**Comment by seabbs** [comment]
> Worth noting that this will give the effective  as opposed to unadjusted 

I think we should put this in the paper as a discussion point

**Comment by SamuelBrand1** [comment]
> > Worth noting that this will give the effective  as opposed to unadjusted
> 
> I think we should put this in the paper as a discussion point
> 
> > One docstring small fix.
> 
> Do we have anything in CI etc that tests docstring examples at least run / docstrings are valid? It seems like this is quite important so it would be odd if the julia community didn't have tools for this

https://documenter.juliadocs.org/stable/man/doctests/#Doctesting-as-Part-of-Testing

IIRC we have talked about this but not done it yet.

**Review by SamuelBrand1 — APPROVED** [review]
LGTM

**Pull request merged**